### PR TITLE
Add community library link to readme

### PR DESCRIPTION
**Pull request by arasan01**
## What is the purpose of the PR?

I would like to add a link to the community library we have created.

## Why did this work occur?

I am developing with swift-dependencies as a central part of my product. In this process, there was a lot of protocol-oriented development. So I thought Swift Macros could automatically convert this to Struct-based and realize function-by-function processing changes. We believe this will be of great help to projects with similar challenges.

## Why create a separate library and then do a PR to add the link? Why not add this feature to this repository?

We believe this will help a great many people, but ideally as a pointfreeco library, this conversion is not even necessary. This is why we are creating it as a community library.

## Libraries to be linked in this PR

https://github.com/arasan01/swift-dependencies-extras

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -272,6 +272,7 @@ following projects all build on top of Dependencies:
 
   * [Dependencies Additions](https://github.com/tgrapperon/swift-dependencies-additions): A
     companion library that provides higher-level dependencies.
+  * [Dependencies Extras](https://github.com/arasan01/swift-dependencies-extras): Macros that make Swift Dependencies more useful
```
Comment (arasan01) [inline README.md:275]:
```suggestion
  * [Dependencies Extras](https://github.com/arasan01/swift-dependencies-extras): Macros that make swift-dependencies more useful
```

File: README.md [hunk 2]
```diff
@@ -272,6 +272,7 @@ following projects all build on top of Dependencies:
 
   * [Dependencies Additions](https://github.com/tgrapperon/swift-dependencies-additions): A
     companion library that provides higher-level dependencies.
+  * [Dependencies Extras](https://github.com/arasan01/swift-dependencies-extras): Macros that make swift-dependencies more useful
```
Comment (stephencelis) [inline README.md:275]: Definitely open to adding this, but given that we already have `swift-dependencies-additions` being a library extending `swift-dependencies` with higher-level dependencies, have you thought about naming the library more specifically related to the macros it offers?

I know renaming a library is a tall order, so no worries if you're happy with the way things are. Just lemme know and I'll merge when you're ready :)

Comment (arasan01) [inline README.md:275]:
```suggestion
  * [Dependencies Protocol Extras](https://github.com/arasan01/swift-dependencies-extras): Library to make swift-dependencies even more useful when using Protocol
```

**Review by mbrandonw — APPROVED** [review]
Thanks @arasan01!

**Pull request merged**